patterns/structural/3-tier.py

```diff
@@ -3,68 +3,77 @@
 Separates presentation, application processing, and data management functions.
 """
 
+from typing import Dict, KeysView, Optional, Type, TypeVar, Union
+
+T = TypeVar("T")
+
 
 class Data:
     """ Data Store Class """
 
     products = {
-        'milk': {'price': 1.50, 'quantity': 10},
-        'eggs': {'price': 0.20, 'quantity': 100},
-        'cheese': {'price': 2.00, 'quantity': 10},
+        "milk": {"price": 1.50, "quantity": 10},
+        "eggs": {"price": 0.20, "quantity": 100},
+        "cheese": {"price": 2.00, "quantity": 10},
     }
 
-    def __get__(self, obj, klas):
+    def __get__(
+        self, obj: Optional[T], klas: Type[T]
```

Comment:
This is an example where I think types add more noise than value. I prefer to keep the code as is, since `__get__` is a standard special function and anyone can look it up

Reply:
We can do partial type hints then. Like ignoring hints for `__get__` while keeping it for the rest of the code.